### PR TITLE
Samples reorg

### LINKED ISSUE #555: Better repo structure for FPC examples
**Description**

As identified in a discussion in #539, the current repo structure lacks of a central place for FPC examples.

As mentioned by @bvavala:
> as we are adding FPC applications/tests, I am noticing that we are fragmenting tests and examples among the integration, examples and, more recently, client_sdk folder. Is there a better way to architect the code/repo?

Let me propose the following structure.
- `$FPC_PATH/examples/chaincode`
- `$FPC_PATH/examples/application`
- `$FPC_PATH/examples/deployment`
- `$FPC_PATH/examples/demos`

We move all our example chaincodes current located in `examples` to `$FPC_PATH/examples/chaincode`. The client sdk samples in `client_sdk/go/sample/` move into `$FPC_PATH/examples/application`, and the k8s tutorial as well as the test-network tutorial, currently located in `$FPC_PATH/integration`, move to `$FPC_PATH/examples/deployment`.

In `$FPC_PATH/examples/demos` we could place more advanced examples like our spectrum auction comprising chaincodes, applications, and deployments.

Also note that the examples in `$FPC_PATH/examples/deployment` should only depend on existing chaincodes and applications.

This new structure with a central folder for all examples and tutorials will enhance FPC users experience.

@bvavala WDYT?

### PR DESCRIPTION
**Pull request by mbrandenburger**
**What this PR does / why we need it**:

This PR implements the changes as discussed in #555. We introduce `/samples` as a central place to locate all samples, including chaincodes, clients SDK, and deployments. 

**Which issue(s) this PR fixes**:

Fixes #555 

**Special notes for your reviewer**:

**Does this PR introduce a user-facing changes and/or breaks backward compatability?**:

```

### CODE REVIEW

File: samples/chaincode/helloworld/README.md [hunk 1]
```diff
@@ -307,7 +305,7 @@ make[1]: Leaving directory '/home/bcuser/work/src/github.com/hyperledger-labs/fa
 ## Time to test!
 Next step is to test the chaincode by invoking transactions, for which you need a basic Fabric network with a channel. You will use the FPC test framework to bring up a Fabric network in which the helloworld code can be executed as a chaincode _in an SGX enclave_.  The Fabric network contains just a single peer and orderer node as defined in `$FPC_PATH/integration/config/`.  Please note the FPC-specific additions to the `core.yaml` to the standard Fabric configurations.  These are marked as `FPC Addition`;  these enable the integration points with Fabric and have to be replicated if you want to use your own Fabric configuration.
 
-Create a file `test.sh` in `$FPC_PATH/examples/helloworld` folder as follows.  Note that the initial lines in the script points to files and folders in FPC framework.
+Create a file `test.sh` in `$FPC_PATH/samples/chaincode/helloworld` folder as follows.  Note that the initial lines in the script points to files and folders in FPC framework.
```
Comment (munapower) [inline samples/chaincode/helloworld/README.md:308]: why is the code for test.sh defined as Makefile?

File: client_sdk/go/README.md [hunk 2]
```diff
@@ -2,10 +2,8 @@
...
-and a [sample program](./sample/main.go) to illustrate it use.
-<!-- TODO: replace above reference to sample program with hello world example once done ...-->
+and a [sample applications](../../samples/application/) to illustrate it use.
+The FPC [Helloworld Tutorial](../../samples/chaincode/helloworld) also demonstrates the usage of the FPC Client SDK for Go.
 
 ## Testing
-See [`sample/main.go`](./sample/main.go) for a simple test program and
-[integration/test-network/README.md](../../integration/test-network/README.md)
-for instructions to run it.
+Before running tests, please make sure you have built the chaincode samples (i.e., run `make -C $FPC_PATH/samples/chaincode`) as they are used for testing.
```
Comment (munapower) [inline client_sdk/go/README.md:9]: ran this make and it worked

**Review by munapower — COMMENTED** [review]
This PR looks good so far I can tell; haven't had the chance so far to test it completely locally if some of these changes do not break the tutorials

File: samples/chaincode/helloworld/README.md (hunk 1 above)
Comment (mbrandenburger) [reply samples/chaincode/helloworld/README.md:308]: good catch

**Review by bvavala — APPROVED** [review]

**Pull request merged**